src/services/geckoTermial.ts

```diff
@@ -0,0 +1,97 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
+
+type SearchResponse = {
+  data: {
+    id: string
+    attributes: {
+      pools: {
+        address: string
+        reserve_in_usd: string
+        tokens: [{ symbol: string }, { symbol: string }]
+        network: {
+          identifier: string
+        }
+      }[]
+    }
+  }
+}
+
+interface CandleResponse {
+  data: Array<{
+    dt: string
+    o: number
+    h: number
+    l: number
+    c: number
+    v: number
+  }>
+  meta?: {
+    noData: boolean
+  }
+}
+
+interface CandleParams {
+  token_id: string
+  pool_id: string
+  resolution: number | string
+  from: string | number
+  to: string | number
+  for_update: boolean
+  currency: 'usd' | 'token'
+  count_back?: number
+}
+
+interface Token {
+  id: string
+  attributes: {
+    name: string
+    symbol: string
+  }
+}
+
+export interface PoolResponse {
+  data: {
+    id: string
+    attributes: {
+      address: string
+      name: string
+      base_token_id: string
+    }
+  }
+  included: [Token, Token]
+}
+
+const geckoTerminalApi = createApi({
+  reducerPath: 'geckoTerminalApi',
+  baseQuery: fetchBaseQuery({
+    // TODO(viet-nv): check prod env
```

Comment:
todo

Reply:
waiting for geckoterminal open CORS for us
<img width="612" alt="image" src="https://user-images.githubusercontent.com/11427213/212237669-65ad6793-a957-465b-a5fc-0a5f67e1f438.png">